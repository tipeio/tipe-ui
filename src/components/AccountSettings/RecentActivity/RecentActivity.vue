<template>
  <accordian text="Recent Activity">
    <activity-item 
      v-for="item in items" 
      :key="item.id" 
      :item="item"
    />
  </accordian>
</template>

<script>
import ActivityItem from './ActivityItem'
import Accordian from '../../Accordian'

export default {
  name: 'RecentActivity',
  components: {
    ActivityItem,
    Accordian
  },
  data: function() {
    return {
      items: [
        {
          title: 'Sam Smith update api key',
          time: 'a moment ago',
          id: '47357838'
        },
        {
          title: 'Sarah Scott created a document',
          time: 'one hour ago',
          id: '5389345'
        }
      ]
    }
  }
}
</script>
