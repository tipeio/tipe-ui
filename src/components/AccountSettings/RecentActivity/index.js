export { default } from './RecentActivity.vue'
export ActivityItem from './ActivityItem.vue'
