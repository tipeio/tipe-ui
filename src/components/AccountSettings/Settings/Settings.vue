<template>
  <accordian text="Settings"><tipe-switch 
    text="File Sharing" 
    sublabel="Turn on to share your file changes and updates" /><tipe-switch text="Backup" /></accordian>
</template>

<script>
import Accordian from '../../Accordian'
import TipeSwitch from '../../Switch'

export default {
  name: 'TipeSettings',
  components: {
    Accordian,
    TipeSwitch
  },
  data: function() {
    return {}
  }
}
</script>
