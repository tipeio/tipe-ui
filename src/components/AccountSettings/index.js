export { default } from './AccountSettings.vue'
