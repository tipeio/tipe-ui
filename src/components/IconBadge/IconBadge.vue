<template>
  <div class="card">
    <tipe-icon
      :icon="icon"
      class="icon"
      height="1.25rem"
      width="1.25rem"
    />
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../Icon'

export default {
  name: 'IconBadge',
  components: {
    TipeIcon
  },
  props: {
    icon: vueTypes.string.isRequired
  }
}
</script>

<style scoped lang="postcss">
.card {
  width: 3.4375rem;
  height: 3.4375rem;
  border-radius: 4px;
  background-color: var(--purple-transparent);
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
