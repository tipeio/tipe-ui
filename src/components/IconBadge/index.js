export { default } from './IconBadge.vue'
