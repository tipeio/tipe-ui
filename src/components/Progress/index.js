export { default } from './Progress.vue'
