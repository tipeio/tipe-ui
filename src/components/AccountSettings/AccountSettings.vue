<template>
  <div class="content">
    <div class="row">
      <icon-badge icon="file" />
      <space-progress 
        :used-space="8.08" 
        :total-space="40" 
        unit="GB" />
    </div>
    <accordian text="Folder Details"/>
    <accordian text="Settings"><tipe-switch text="File Sharing" /><tipe-switch text="Backup" /></accordian>
    <accordian text="Recent Activity"><activity-item 
      title="Sam Smith update api key" 
      time="a moment ago"/><activity-item 
        title="Sam Smith update api key" 
        time="a moment ago"/></accordian>
  </div>
</template>

<script>
import IconBadge from '../IconBadge'
import Accordian from '../Accordian'
import SpaceProgress from '../SpaceProgress'
import TipeSwitch from '../Switch'
import ActivityItem from '../RecentActivity/ActivityItem'

export default {
  name: 'AccountSettings',
  components: {
    IconBadge,
    Accordian,
    SpaceProgress,
    TipeSwitch,
    ActivityItem
  }
}
</script> 

<style scoped lang="postcss">
.content {
  width: 252px;
  display: grid;
  grid-row-gap: 1.5rem;
}

.row {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-around;
}
</style>
