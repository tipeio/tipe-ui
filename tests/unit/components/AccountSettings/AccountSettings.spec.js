import { shallowMount } from '@vue/test-utils'
import AccountSettings from '@/components/AccountSettings'
import RecentActivity from '@/components/AccountSettings/RecentActivity'
import Settings from '@/components/AccountSettings/Settings'
import IconBadge from '@/components/IconBadge'
import Progress from '@/components/Progress'

describe('Settings.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper).toMatchSnapshot()
  })
  it('should have the pannel content class', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper.classes()).toContain('panel-content')
  })
  it('should render the icon badge component', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper.contains(IconBadge)).toBe(true)
  })
  it('should render the progress component', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper.contains(Progress)).toBe(true)
  })
  it('should render the recent activity component', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper.contains(RecentActivity)).toBe(true)
  })
  it('should render the settings component', () => {
    const wrapper = shallowMount(AccountSettings)
    expect(wrapper.contains(Settings)).toBe(true)
  })
})
