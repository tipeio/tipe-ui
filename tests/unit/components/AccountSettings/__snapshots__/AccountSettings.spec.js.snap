// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings.vue renders 1`] = `
<div class="panel-content">
  <div class="row">
    <IconBadge-stub icon="rocket" height="20px" width="20px"></IconBadge-stub>
    <TipeProgress-stub total="40" used="8.08" unit="GB" title="Space"></TipeProgress-stub>
  </div>
  <Accordian-stub text="Folder Details"></Accordian-stub>
  <TipeSettings-stub></TipeSettings-stub>
  <RecentActivity-stub></RecentActivity-stub>
</div>
`;
